packages/@lwc/synthetic-shadow/src/faux-shadow/events.ts

```diff
@@ -59,101 +58,19 @@ function getRootNodeHost(node: Node, options: GetRootNodeOptions): Node {
-    const originalTarget = eventTargetGetter.call(this);
-    const composedPath = pathComposer(originalTarget, this.composed);
-    const doc = getOwnerDocument(originalTarget as Node);
-
-    // Handle cases where the currentTarget is null (for async events),
-    // and when an event has been added to Window
-    if (!(originalCurrentTarget instanceof Node)) {
-        // TODO [#1511]: Special escape hatch to support legacy behavior. Should be fixed.
-        // If the event's target is being accessed async and originalTarget is not a keyed element, do not retarget
-        if (isNull(originalCurrentTarget) && isUndefined(getNodeOwnerKey(originalTarget as Node))) {
-            return originalTarget;
-        }
-        return retarget(doc, composedPath);
-    } else if (originalCurrentTarget === doc || originalCurrentTarget === doc.body) {
-        // TODO [#1530]: If currentTarget is document or document.body (Third party libraries that have global event listeners)
-        // and the originalTarget is not a keyed element, do not retarget
-        if (isUndefined(getNodeOwnerKey(originalTarget as Node))) {
-            return originalTarget;
-        }
-        return retarget(doc, composedPath);
-    }
-
-    let actualCurrentTarget = originalCurrentTarget;
-    let actualPath = composedPath;
-
-    // Address the possibility that `currentTarget` is a shadow root
-    if (isHostElement(originalCurrentTarget)) {
-        const context = eventToContextMap.get(this);
-        if (context === EventListenerContext.SHADOW_ROOT_LISTENER) {
-            actualCurrentTarget = getShadowRoot(originalCurrentTarget);
-        }
-    }
-
-    // Address the possibility that `target` is a shadow root
-    if (isHostElement(originalTarget) && eventsDispatchedDirectlyOnShadowRoot.has(this)) {
-        actualPath = pathComposer(getShadowRoot(originalTarget), this.composed);
-    }
-
-    return retarget(actualCurrentTarget, actualPath);
-}
-
-function composedPathValue(this: Event): EventTarget[] {
-    const originalTarget = eventTargetGetter.call(this);
-    const originalCurrentTarget = eventCurrentTargetGetter.call(this);
-
-    // If the event has completed propagation, the composedPath should be an empty array.
-    if (isNull(originalCurrentTarget)) {
-        return [];
-    }
-
-    // Address the possibility that `target` is a shadow root
-    let actualTarget = originalTarget;
-    if (isHostElement(originalTarget) && eventsDispatchedDirectlyOnShadowRoot.has(this)) {
-        actualTarget = getShadowRoot(originalTarget);
-    }
-
-    return pathComposer(actualTarget, this.composed);
-}
-
 export function doesEventNeedPatch(e: Event): boolean {
     const originalTarget = eventTargetGetter.call(e);
     return originalTarget instanceof Node && isNodeDeepShadowed(originalTarget);
 }
 
+const patchedEvents: WeakSet<Event> = new WeakSet();
+
 export function patchEvent(event: Event) {
-    if (eventToContextMap.has(event)) {
```

Comment:
I would expect that more code will have to change once https://github.com/salesforce/lwc/pull/2111 is merged.

Reply:
This PR is actually going in first.